src/osm_graph_builder.cpp

```diff
@@ -203,6 +219,8 @@ OSMRoutingGraph load_osm_routing_graph_from_pbf(
 					unsigned routing_id_of_last_routing_node = routing_node.to_local(node_list[0]);
 
 					double dist_since_last_routing_node = 0;
+					std::vector<float>modelling_node_latitude;
```

Comment:
Please move the variable to the top level scope of the function and use vector clear here. This avoids allocating operator new per way.